datasets/jokes.py

```diff
@@ -53,7 +56,7 @@ def generate_examples(self, num_examples):
                 selected_jokes.append(joke)
                 is_question.append(False)
                 time_jump = create_time_jump(self.minutes_low, self.minutes_high)
-                waits.append(WaitCreator.create_wait(tokens=self.filler_tokens, time=time_jump))
+                waits.append(WaitCreator.create_wait(tokens=filler_tokens, time=time_jump))
```

Comment:
The semantics of waits require all conditions to be met, so here this would mean that the test should wait until `filler_tokens` have benn exchanged AND `time_jump` has passed. Time jumps are now trickier because they could violate the limits of the memory span, but maybe here we can do with just token waits.

Reply:
I think its fine... can you construct an instance where its not fine? My thinking is:
- For the token component, that competes with all the other tests as normal, as other test run, this token wait decreases relative to the total tokens elapsed.
- After the token wait is done, then next search for an id will check the time blocked tests first and return them.

Reply:
Running it in the debugger, yeah the time is fast forwarded first. because it resolves the time waits independent of the tokens waits.

Reply:
I might be overzealous about this, but if one test has a wait on time, and there are sufficient tests to run until that time mark comes, then the scheduler will run those first. In that case, the maximum memory span might be exceeded.